src/Util/InvalidArgumentHelper.php

```diff
@@ -22,11 +22,12 @@ public static function factory(int $argument, string $type, $value = null): Exce
 
         return new Exception(
             \sprintf(
-                'Argument #%d%sof %s::%s() must be a %s',
+                'Argument #%d%sof %s::%s() must be %s %s',
                 $argument,
                 $value !== null ? ' (' . \gettype($value) . '#' . $value . ')' : ' (No Value) ',
                 $stack[1]['class'],
                 $stack[1]['function'],
+                \in_array($type[0], ['a', 'e', 'i', 'o', 'u']) ? 'an' : 'a',
```

Comment:
Can you check if `$type` is already guaranteed lowercase here? Just to be sure and prevent possible edge cases like `must be a Iterator` or `must be a Object`.